### PR TITLE
docs: add documentation for UTM module

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Documents the UTM module (just because is the simplest!)

# To Test
- Verify the README --> https://github.com/cowprotocol/cowswap/blob/docs/utm-module/src/modules/utm/utm-module.md
- Check the changes in the overall README --> now it has a link to the UTM docs, and some structure to add the module links as we document it https://github.com/cowprotocol/cowswap/blob/docs/utm-module/docs/architecture-overview.md
- Give feedback if you feel this is a good way to document what a module does. This didn't need much documentation, so I was just explained it don't depend on other modules, and what's the facade. Other modules might incorporate some additional things (like screenshots, sequence diagrams, etc). I think we should be reasonable and not add for the sake of adding, just to make it clear for someone to understand

image

### CODE REVIEW

**Comment by shoom3301** [comment]
Nevertheless, I still think, that all analytics stuff (including UTM) should be handled by [GTM](https://tagmanager.google.com/).

File: src/modules/utm/index.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
-export * from './hooks'
-export * from './types'
+export { useInitializeUtm, useUtm } from './hooks'
+export type { UtmParams } from './types'
```
Comment (alfetopito) [inline src/modules/utm/index.ts:2]: Why change the export?
This makes it behave worse IMO as every new export has to manually be added here

File: src/modules/utm/utm-module.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+See also:
+
+- [CoW Swap Architecture](../../../docs/architecture-overview.md)
+- [Broader CoW Swap Architecture](https://github.com/cowprotocol/cowswap-diagrams)
+
+![UTM module](./utm-module.drawio.svg)
+
+## Desciption
+
+This module handles adds support for handling UTM codes.
```
Comment (alfetopito) [inline src/modules/utm/utm-module.md:12]:
```suggestion
This module adds support for handling UTM codes.
```

File: src/modules/utm/utm-module.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+For creating campagins or collaborations, there's a process:
+
+- [Create UTM Codes for on-chain analytics](https://www.notion.so/cownation/Create-UTM-codes-for-on-chain-analytics-cd1ec4ce33964da48a374c8d17b00913?pvs=4)
+
+# Details
+
+This module will expose two hooks:
+
+- `useUtm()`: Exposes the UTM codes
+- `useInitializeUtm()`: Hook that will make sure the UTM codes are intialised bade on the URL parameters
```
Comment (alfetopito) [inline src/modules/utm/utm-module.md:30]: Where's the interface for changing utm values?

**Review by alfetopito — APPROVED** [review]

**Comment by anxolin** [comment]
Related: https://www.notion.so/cownation/Architecture-FAQ-Must-read-599a8c8907154721b3d98f1ced4fc829?pvs=4

File: src/modules/utm/index.ts (hunk 1 above)
Comment (anxolin) [reply src/modules/utm/index.ts:2]: I think is better to cherry-pick, my hope it is really a reduce number of items the ones exported

**Pull request merged**